PsfLauncher/PsfPowershellScriptRunner.h

```diff
@@ -139,11 +162,86 @@ class PsfPowershellScriptRunner
 		return scriptStruct;
 	}
 
+	std::wstring ReplacePsuedoRootVariables(std::wstring inString, std::filesystem::path packageRoot, std::filesystem::path packageWritableRoot)
+	{
+		//Allow for a substitution in the strings for a new pseudo variable %MsixPackageRoot% so that arguments can point to files
+		//inside the package using a syntax relative to the package root rather than rely on VFS pathing which can't kick in yet.
+		std::wstring outString = inString;
+		std::wstring var2rep1 = L"%MsixPackageRoot%";
+		std::wstring var2rep2 = L"%MsixWritablePackageRoot%";
+		
+		std::wstring::size_type pos1 = 0u;
+		std::wstring repargs1 = packageRoot.c_str();
+		while ((pos1 = outString.find(var2rep1, pos1)) != std::string::npos) {
+			outString.replace(pos1, var2rep1.length(), repargs1);
+			pos1 += repargs1.length();
+		}
+		std::wstring::size_type pos2 = 0u;
+		std::wstring repargs2 = packageWritableRoot.c_str();
+		while ((pos2 = outString.find(var2rep2, pos2)) != std::string::npos) {
+			outString.replace(pos2, var2rep2.length(), repargs2);
+			pos2 += repargs2.length();
+		}		
+		return outString;
+	}
+	std::wstring Dequote(std::wstring inString)
+	{
+		// Remove quotation-like marks around edges of a string reference to a file, if present.
+		if (inString.length() > 2)
+		{
+			if (inString[0] == L'\"' && inString[inString.length() - 1] == L'\"')
+			{
+				return inString.substr(1, inString.length() - 2);
+			}
+			if (inString[0] == L'\'' && inString[inString.length() - 1] == L'\'')
+			{
+				return inString.substr(1, inString.length() - 2);
+			}
+			if (inString.length() > 4)
+			{
+				// Check for Powershell style references too
+				if (inString[0] == L'`' && inString[1] == L'\'' && inString[inString.length() - 2] == L'`' && inString[inString.length() - 1] == L'\'')
+				{
+					return inString.substr(2, inString.length() - 4);
+				}
+				if (inString[0] == L'`' && inString[1] == L'\"' && inString[inString.length() - 2] == L'`' && inString[inString.length() - 1] == L'\"')
+				{
+					return inString.substr(2, inString.length() - 4);
+				}
+			}
+		}
+		return inString;
+	}
+	std::wstring EscapeFilename4PowerShell(std::filesystem::path inputPath)

```

Comment:
Please replace 4 with Four.

Reply:
?? Not sure what the request here is. Are you suggesting to add a variable called "Four" with value 4 to replace use of 4 in the code around this area?

Reply:
Oops.  I meant to replace 4 with For, not Four.

The varible name should be ExcapeFilenameForPowerShell.